### ISSUE TITLE
Meteor build process halts on this valid JS

### ISSUE THREAD

**Issue by jorisroling** [meteor/meteor#10002]
It took me half a day to figure this one out...

The following code halts the meteor build process with a arbitrary error
```javascript
    if (globals.never) {
      const b = new Date();
    }
```

But the following code is happily build, no error
```javascript
    if (globals.never) {
      new Date();
    }
```

I know the `const b` is never used, but crash on it without giving a proper clue?

**Comment by benjamn** [comment]
I can't reproduce this. What version of Meteor are you using? Do you have any plugins in a `.babelrc` file? Can you provide a small reproduction?

**Comment by jorisroling** [comment]
I have put a little repo online here: https://github.com/jorisroling/simple-todos-showing-error
It is a `meteor create simple-todos` with the lines above in `client/main.js`

**Comment by tcastelli** [comment]
Could be related with #9976 , although in our case we have looked for unused const declarations and there was none

**Comment by klaussner** [comment]
This issue is fixed in the Meteor 1.7.1 beta, thanks to #10042/#10045.
